src/components/datagrid/data_grid.tsx

```diff
@@ -598,37 +598,29 @@ export const EuiDataGrid: FunctionComponent<EuiDataGridProps> = props => {
     </EuiI18n>
   );
 
-  const [cellsUpdateFocus, setCellsUpdateFocus] = useState<
-    Array<Function[] | null[]>
-  >([]);
+  const [cellsUpdateFocus] = useState<Map<string, Function>>(new Map());
 
   const updateFocus = (focusedCell: [number, number]) => {
-    const updateFocus = cellsUpdateFocus[focusedCell[0]][focusedCell[1]];
-
-    if (updateFocus) {
-      updateFocus();
+    const key = `${focusedCell[0]}-${focusedCell[1]}`;
+    if (cellsUpdateFocus.has(key)) {
+      requestAnimationFrame(() => {
+        cellsUpdateFocus.get(key)!();
+      });
     }
   };
 
   const datagridContext = {
     onFocusUpdate: (cell: [number, number], updateFocus: Function) => {
       if (pagination) {
-        // Receives the row index as for the whole set
-        // and normalizes it for the visible rows in the grid
-        const pageIndex = pagination.pageIndex;
-        const pageSize = pagination.pageSize;
-        const rowIndex = Math.ceil(cell[1] - pageIndex * pageSize);
-
-        if (!cellsUpdateFocus[cell[0]]) {
-          cellsUpdateFocus[cell[0]] = [];
-        }
-
-        cellsUpdateFocus[cell[0]][rowIndex] = updateFocus;
+        const key = `${cell[0]}-${cell[1]}`;
 
-        setCellsUpdateFocus(cellsUpdateFocus);
+        // this intentionally and purposefully mutates the existing `cellsUpdateFocus` object as the
```

Comment:
To be honest I don't know of The React Way to manage this interaction. I'll think more about it, but this is clean enough to merge in. There's a similar situation in the datagrid's in-memory rendering & tracking of cell values. Likely, we need to find a better solution for all of EUI than abusing `requestAnimationFrame` in these circumstances.